### PR TITLE
Fix build with new glib versions

### PR DESCRIPTION
**Pull request by z3ntu**
> Replace use of deprecated g_type_class_add_private.
> Replace use of deprecated G_TYPE_INSTANCE_GET_PRIVATE.
> 
> Bump the minimal required version of glib to 2.38.0, the version
> where G_PRIVATE_ADD was added.

Also
> Replace gnome-common in autogen.sh
> See https://wiki.gnome.org/Projects/GnomeCommon/Migration

### CODE REVIEW

File: autogen.sh [hunk 1]
```diff
@@ -1,11 +1,41 @@
...
+        echo "" >&2
+fi
+
+aclocal --install || exit 1
+intltoolize --force --copy --automake || exit 1
+gtkdocize --copy || exit 1
+autoreconf --verbose --force --install || exit 1
+
+cd "$olddir"
+if [ "$NOCONFIGURE" = "" ]; then
```
Comment (peat-psuwit) [inline autogen.sh:31]: You have to also update `debian/rules`; it currently uses `NO_CONFIGURE`.

**Review by peat-psuwit — CHANGES_REQUESTED** [review]

File: autogen.sh (hunk 1 above)
Comment (z3ntu) [reply autogen.sh:31]: Good catch, fixed

**Review by peat-psuwit — APPROVED** [review]
Tested with most of the tests from [Ubuntu's test plan](https://wiki.ubuntu.com/Process/Merges/TestPlan/indicator-messages). Seems to work fine. Did not test the Unity7 part and the "Show Messages on Greeter" settings.

**Comment by peat-psuwit** [comment]
For QA: see if the message indicator works correctly. See if translations are coming through.

**Comment by UniversalSuperBox** [comment]
The indicator-messages test plan succeeded on my Xperia X.

**Pull request merged**